test/CMakeLists.txt

```diff
@@ -85,119 +86,177 @@ list(APPEND TESTS connect_dend)
+        ${CMAKE_COMMAND} -E env COVERAGE_FILE=.coverage.coreneuron_fornetcon_py ${PYTHON_EXECUTABLE}
+        -m pytest --cov-report=xml --cov=neuron
+        ${PROJECT_SOURCE_DIR}/test/coreneuron/test_fornetcon.py
+      WORKING_DIRECTORY ${PROJECT_BINARY_DIR}/test/mod)
+    add_test(
+      NAME coreneuron_datareturn_py
+      COMMAND
+        ${CMAKE_COMMAND} -E env COVERAGE_FILE=.coverage.coreneuron_datareturn_py
+        ${PYTHON_EXECUTABLE} -m pytest --cov-report=xml --cov=neuron
+        ${PROJECT_SOURCE_DIR}/test/coreneuron/test_datareturn.py
+      WORKING_DIRECTORY ${PROJECT_BINARY_DIR}/test/mod)
+    add_test(
+      NAME coreneuron_test_units_py
+      COMMAND
+        ${CMAKE_COMMAND} -E env COVERAGE_FILE=.coverage.coreneuron_test_units_py
+        ${PYTHON_EXECUTABLE} -m pytest --cov-report=xml --cov=neuron
+        ${PROJECT_SOURCE_DIR}/test/coreneuron/test_units.py
+      WORKING_DIRECTORY ${PROJECT_BINARY_DIR}/test/mod)
+    add_test(
+      NAME coreneuron_test_natrans_py
+      COMMAND
+        ${CMAKE_COMMAND} -E env COVERAGE_FILE=.coverage.coreneuron_test_natrans_py
+        ${PYTHON_EXECUTABLE} -m pytest --cov-report=xml --cov=neuron
+        ${PROJECT_SOURCE_DIR}/test/gjtests/test_natrans.py
+      WORKING_DIRECTORY ${PROJECT_BINARY_DIR}/test/mod)
+    list(
+      APPEND TESTS
+             coreneuron_direct_py
+             coreneuron_direct_hoc
+             coreneuron_spikes_py
+             coreneuron_spikes_file_mode_py
+             coreneuron_fornetcon_py
+             coreneuron_datareturn_py
+             coreneuron_test_units_py
+             coreneuron_test_natrans_py)
     if(NRN_ENABLE_MPI)
       find_python_module(mpi4py)
       if(mpi4py_FOUND)
         add_test(
-                NAME coreneuron_spikes_mpi_py  COMMAND ${MPIEXEC} -np 2 ${PYTHON_EXECUTABLE}
-                                              ${PROJECT_SOURCE_DIR}/test/coreneuron/test_spikes.py mpi4py
-                                      WORKING_DIRECTORY ${PROJECT_BINARY_DIR}/test/mod)
+          NAME coreneuron_spikes_mpi_py
+          COMMAND
+            ${MPIEXEC} ${MPIEXEC_NUMPROC_FLAG} 2 ${MPIEXEC_PREFLAGS} ${PYTHON_EXECUTABLE}
+            ${MPIEXEC_POSTFLAGS} ${PROJECT_SOURCE_DIR}/test/coreneuron/test_spikes.py mpi4py
+          WORKING_DIRECTORY ${PROJECT_BINARY_DIR}/test/mod)
         list(APPEND TESTS coreneuron_spikes_mpi_py)
       endif()
-        # using -pyexe to workaround #894
-        add_test(
-                NAME coreneuron_spikes_mpi_file_mode_py  COMMAND ${MPIEXEC} -np 2 ${CMAKE_HOST_SYSTEM_PROCESSOR}/special
-                                              -python -pyexe ${PYTHON_EXECUTABLE} ${PROJECT_SOURCE_DIR}/test/coreneuron/test_spikes.py nrnmpi_init file_mode
-                                      WORKING_DIRECTORY ${PROJECT_BINARY_DIR}/test/mod)
-        list(APPEND TESTS coreneuron_spikes_mpi_file_mode_py)
+      # Using -pyexe was a first workaround for #894, but it seems that we also need to avoid using
```

Comment:
```suggestion
      # Using -pyexe was a first workaround for Github issue #894, but it seems that we also need to avoid using
```

Reply:
Done in 6fc0e7bbbd33ccd864e93723885ad3ecd8a621e5.